### ISSUE TITLE
RandomCrop explicit padding still triggers oversized matrix scaling

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4542]
RandomCrop rescales its transformation matrix against the unpadded input when the requested size exceeds either input axis and `pad_if_needed=False`. Explicit padding can already make the crop fit, but the correction still runs. Slice-mode image coordinates then disagree with keypoints/boxes; resample mode also distorts the image.

Reproduced on PR #4538 at `b9891a44`, Linux CPU, PyTorch 2.14.0:

```python
import torch
import kornia.augmentation as K
from kornia.geometry.keypoints import Keypoints

image = torch.arange(9.).reshape(1, 1, 3, 3)
crop = K.RandomCrop((4, 4), padding=1, p=1, cropping_mode="slice")
params = crop.forward_parameters(image.shape)
params["src"] = torch.tensor([[[0., 0.], [3., 0.], [3., 3.], [0., 3.]]])
output = crop(image, params=params)
points = Keypoints(torch.tensor([[[2., 2.]]]))
print(crop.transform_matrix)
print(crop.transform_keypoints(points, params, crop.flags, crop.transform_matrix).data)
```

The matrix is `diag(4/3, 4/3, 1)` and the keypoint becomes `(4, 4)`, outside the 4x4 output. The image pixel remains at `(3, 3)` after pre-padding. The crop matrix should be identity for these explicit source/destination corners; pre-padding already supplies the coordinate shift. With `cropping_mode="resample"`, the same incorrect scaling changes the bottom-right ramp value from 8 to 5.

Related: #4414 covers oversized crops without padding. PR #4538 will document and pin this separate explicit-padding case without changing runtime behavior.

Written by Codex (gpt-6-astra) on behalf of @ducha-aiki
